text/0000-Bundle-Load-API.md

```diff
@@ -0,0 +1,135 @@
+- Start Date: 2019-07-26
+- RFC PR: https://github.com/CppMicroServices/rfcs/pull/7
+- CppMicroServices Issue: https://github.com/CppMicroServices/CppMicroServices/issues/376
+
+# Add Load API in Bundle class 
+
+## Summary
+
+Currently there is no API function in Bundle class to load bundles/dynamic libraries at runtime.
+This document discusses need for such API in the particular use case of declarative services (referred as DS henceforth in this document) and its possible design/solution.
+
+## Motivation
+
+Developers using DS rely on DS's service component runtime to delay/lazily load the bundles.
+Since DS hides all the boilerplate code of bundle lifetime (install => start => stop => uninstall), service consumers can not directly use framework APIs (e.g. Start) to load bundles
+
+Moverover, one could argue that framework's Start() function could be modified to accomodate this functionality instead of having a new member function in the Bundle class.
+However, this not possible as DS runtime is designed such that it only loads the service class constructors/destructors by calling extern "C" functions instead of framework's Start function.
+As such, framework's Start API can not solve this specific use case for DS.
+
+Because of these reasons, DS uses OS specific dlopen/LoadLibraryW calls for loading bundles/shared objects.
+
+This loading of libraries/bundles increases the overhead in the declarative services while doing the symbol resoution and ref count increment done in the system.
+
+Adding to that, this functionality to load bundles/shared objects is currently implemented at two places
+
+1) Declarative services runtime
+2) framework
+
+During the application run, these different callsites may become out of sync.
+
+Given this it seems appropriate to have a Bundle::BundleLoad API for CppMicroServices library.
+
+## Requirements
+
+For the specific use case of DS (and similar extender clients) ,
+
+1) implement Bundle::LoadBundle such that it encapsulates shared library loading and symbol resolution, not exposing OS specific data types
```

Comment:
this requirement needs to be refined a little. It reads more like an implementation detail than a requirement. Instead of starting with "implement", try starting with "needs to provide a way" and see where that leads the wording of the requirement.